ste/getProxy_windows.go

```diff
@@ -0,0 +1,35 @@
+package ste
+
+import (
+	"golang.org/x/net/http/httpproxy"
+	"golang.org/x/sys/windows/registry"
+	"net/http"
+	"net/url"
+)
+
+func getProxy() func(*http.Request) (*url.URL, error) {
+	if prox := httpproxy.FromEnvironment(); (prox.HTTPProxy != "" || prox.HTTPSProxy != "") && prox.NoProxy == "" {
```

Comment:
Why does this require that NoProxy == "".  That seems like a bug to me. 

NoProxy is a list of addresses for which no proxy should be used.  If that list is populated, that doesn't appear to be a reason to disregard the "from environment" proxy.  IMHO we should use the environment-based proxy if either HTTPProxy or HTTPSProxy is set.

Reply:
This was a misunderstanding of the environment variable's purpose.